### PR TITLE
quic: use oghttp2 header validator to validate request/response headers

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Commit Message: switch to use `http2::adapter::HeaderValidator` from QUICHE to validate request/response header names and values. This library validate both header name and value to protect against header smuggling attack.

Additional Description: this change also fix a bug where `override_stream_error_on_invalid_http_message` didn't take effect on invalid headers rejected by EnvoyQuicStream::validateHeader() .

Risk Level: low
Testing: added integration tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Review by alyssawilk — DISMISSED** [review]
code looks good but CI doesn't.  PTAL?

**Comment by danzh2010** [comment]
> code looks good but CI doesn't. PTAL?

Fixed!

File: test/integration/quic_http_integration_test.cc [hunk 1]
```diff
@@ -704,41 +704,6 @@ TEST_P(QuicHttpIntegrationTest, CertVerificationFailure) {
   EXPECT_EQ(failure_reason, codec_client_->connection()->transportFailureReason());
 }
 
-// HTTP3 doesn't support 101 SwitchProtocol response code, the client should
-// reset the request.
-TEST_P(QuicHttpIntegrationTest, Reset101SwitchProtocolResponse) {
```
Comment (alyssawilk) [inline test/integration/quic_http_integration_test.cc:709]: if http/3 disallows why remove this?

Comment (danzh2010) [reply test/integration/quic_http_integration_test.cc:709]: It's moved to protocol_integration_test

Comment (alyssawilk) [reply test/integration/quic_http_integration_test.cc:709]: Oh PEBKAC sorry, I saw the validation removal and the test "removal" and didn't look more carefully.

**Pull request merged**